lib/__tests/parse.js

```diff
@@ -462,5 +462,40 @@ describe('parse', () => {
                 [189, 4, 79, 'String']
             ]);
         });
+
+        it('should calculate location of layer correctly', () => {
```

Comment:
A separate test case for this isn't necessary. You can add a couple of `@layer` at-rules (one with a single name and another containing `.`) to `fixture/stringify.css` to cover overall parsing correctness, including locations. To update `fixture/stringify.ast`, temporarily uncomment this [line](https://github.com/csstree/csstree/blob/master/lib/__tests/common.js#L26), run `npm test`, and then comment it out again afterward.

Reply:
Got it. 👍